File: data-structures/arrays/vb/array_1D_for_each_loop.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module Program

    ' The Main subroutine is the entry point for all VB programs
    Sub Main()
        ' Declare and initialise the array of words
        Dim spellingWords = New String(4) {"me", "do", "her", "it", "him"}

        ' Output the spelling words array
        DisplayWords(spellingWords)
    End Sub


    ' Display the words from the given spelling list
    Sub DisplayWords(ByVal spellingWords As String())
        Console.WriteLine("Spelling words:")

        ' Repeat for each value in the array
        For Each word In spellingWords
            Console.WriteLine(word)
        Next
    End Sub


End Module

File: data-structures/arrays/python/array_1D_for_each_loop.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def display_words(spelling_words):
    """Display the words from the given spelling list"""

    print("Spelling words:")
    
    # Repeat for each value in the list
    for word in spelling_words:
        print(word)


def main():
    # Declare and initialise the list of words
    spelling_words = ["me", "do", "it", "her", "him"]

    # Output the spelling words list
    display_words(spelling_words)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()
